src/applications/vaos/appointment-list/redux/selectors.js

```diff
@@ -100,13 +102,50 @@ export function selectFutureStatus(state) {
 
 export const selectFutureAppointments = createSelector(
   selectFeatureExpressCare,
+  selectFeatureHomepageRefresh,
+  state => state.appointments.appointmentListFilter,
   state => state.appointments.pending,
   state => state.appointments.confirmed,
-  (showExpressCare, pending, confirmed) => {
+  state => state.appointments.past,
+  (
+    showExpressCare,
+    isSelectFeatureHomepageRefresh,
+    appointmentListFilter,
+    pending,
+    confirmed,
+    past,
+  ) => {
     if (!confirmed || !pending) {
       return null;
     }
 
+    let appointments;
+    if (isSelectFeatureHomepageRefresh) {
+      if (appointmentListFilter === 'upcoming') {
```

Comment:
I think most of this logic would go away by switching to separate pages instead of a single filter, and we can leave it to later tickets to pull in the data how they need it.

Reply:
Create a new branch with only the changes needed. So this file is not modified anymore.